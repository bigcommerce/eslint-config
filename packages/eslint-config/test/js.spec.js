import { join } from 'path';
import { fileURLToPath } from 'url';

import eslintSerializer from './eslint-serializer.js';
import { getEslint } from './getEslint.js';

const __dirname = fileURLToPath(new URL('.', import.meta.url));

expect.addSnapshotSerializer(eslintSerializer);

const fileName = join(__dirname, '__fixtures__', 'file_js.js');

it('keeps rules stable', async () => {
  const eslint = await getEslint();
  const output = await eslint.calculateConfigForFile(fileName);

  expect(output).toMatchSnapshot();
});

it('checks that the rules are working', async () => {
  const eslint = await getEslint();
  const [result] = await eslint.lintFiles([fileName]);

  expect(result.messages).toEqual([]);
});

it('checks for rule deprecations', async () => {
  const eslint = await getEslint();
  const [result] = await eslint.lintFiles([fileName]);

  expect(result.usedDeprecatedRules).toMatchInlineSnapshot(`[]`);
});
