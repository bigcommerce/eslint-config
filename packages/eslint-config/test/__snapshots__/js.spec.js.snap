// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`keeps rules stable 1`] = `
{
  "env": {
    "browser": true,
    "es6": true,
    "node": true,
  },
  "globals": {},
  "ignorePatterns": [],
  "noInlineConfig": undefined,
  "parser": null,
  "parserOptions": {
    "ecmaFeatures": {
      "jsx": true,
    },
    "ecmaVersion": "latest",
    "sourceType": "module",
  },
  "plugins": [
    "@stylistic",
    "switch-case",
    "gettext",
    "import",
    "jsdoc",
    "prettier",
    "react",
    "jsx-a11y",
    "react-hooks",
  ],
  "reportUnusedDisableDirectives": true,
  "rules": {
    "@babel/object-curly-spacing": [
      0,
    ],
    "@babel/semi": [
      0,
    ],
    "@stylistic/padding-line-between-statements": [
      1,
      {
        "blankLine": "always",
        "next": "return",
        "prev": "*",
      },
      {
        "blankLine": "always",
        "next": [
          "block",
          "block-like",
          "class",
          "const",
          "let",
          "var",
          "interface",
        ],
        "prev": "*",
      },
      {
        "blankLine": "always",
        "next": "*",
        "prev": [
          "block",
          "block-like",
          "class",
          "const",
          "let",
          "var",
          "interface",
        ],
      },
      {
        "blankLine": "any",
        "next": [
          "const",
          "let",
          "var",
        ],
        "prev": [
          "const",
          "let",
          "var",
        ],
      },
      {
        "blankLine": "any",
        "next": [
          "case",
        ],
        "prev": [
          "case",
        ],
      },
      {
        "blankLine": "always",
        "next": "*",
        "prev": "directive",
      },
      {
        "blankLine": "any",
        "next": "directive",
        "prev": "directive",
      },
    ],
    "@stylistic/spaced-comment": [
      1,
      "always",
      {
        "block": {
          "balanced": true,
          "exceptions": [
            "-",
            "+",
          ],
          "markers": [
            "=",
            "!",
          ],
        },
        "line": {
          "exceptions": [
            "-",
            "+",
          ],
          "markers": [
            "=",
            "!",
          ],
        },
      },
    ],
    "@typescript-eslint/block-spacing": [
      0,
    ],
    "@typescript-eslint/brace-style": [
      0,
    ],
    "@typescript-eslint/comma-dangle": [
      0,
    ],
    "@typescript-eslint/comma-spacing": [
      0,
    ],
    "@typescript-eslint/func-call-spacing": [
      0,
    ],
    "@typescript-eslint/indent": [
      0,
    ],
    "@typescript-eslint/key-spacing": [
      0,
    ],
    "@typescript-eslint/keyword-spacing": [
      0,
    ],
    "@typescript-eslint/lines-around-comment": [
      0,
    ],
    "@typescript-eslint/member-delimiter-style": [
      0,
    ],
    "@typescript-eslint/no-extra-parens": [
      0,
    ],
    "@typescript-eslint/no-extra-semi": [
      0,
    ],
    "@typescript-eslint/object-curly-spacing": [
      0,
    ],
    "@typescript-eslint/quotes": [
      0,
    ],
    "@typescript-eslint/semi": [
      0,
    ],
    "@typescript-eslint/space-before-blocks": [
      0,
    ],
    "@typescript-eslint/space-before-function-paren": [
      0,
    ],
    "@typescript-eslint/space-infix-ops": [
      0,
    ],
    "@typescript-eslint/type-annotation-spacing": [
      0,
    ],
    "array-bracket-newline": [
      0,
    ],
    "array-bracket-spacing": [
      0,
    ],
    "array-callback-return": [
      2,
    ],
    "array-element-newline": [
      0,
    ],
    "arrow-body-style": [
      0,
      "as-needed",
      {
        "requireReturnForObjectLiteral": false,
      },
    ],
    "arrow-parens": [
      0,
    ],
    "arrow-spacing": [
      0,
    ],
    "babel/object-curly-spacing": [
      0,
    ],
    "babel/quotes": [
      0,
    ],
    "babel/semi": [
      0,
    ],
    "block-scoped-var": [
      2,
    ],
    "block-spacing": [
      0,
    ],
    "brace-style": [
      0,
    ],
    "camelcase": [
      2,
      {
        "ignoreDestructuring": false,
        "ignoreGlobals": false,
        "ignoreImports": false,
        "properties": "never",
      },
    ],
    "comma-dangle": [
      0,
    ],
    "comma-spacing": [
      0,
    ],
    "comma-style": [
      0,
    ],
    "complexity": [
      2,
    ],
    "computed-property-spacing": [
      0,
    ],
    "consistent-return": [
      2,
    ],
    "constructor-super": [
      2,
    ],
    "curly": [
      2,
      "all",
    ],
    "default-case": [
      2,
    ],
    "default-param-last": [
      2,
    ],
    "dot-location": [
      0,
    ],
    "dot-notation": [
      2,
      {
        "allowKeywords": true,
        "allowPattern": "",
      },
    ],
    "eol-last": [
      0,
    ],
    "eqeqeq": [
      2,
      "smart",
    ],
    "flowtype/boolean-style": [
      0,
    ],
    "flowtype/delimiter-dangle": [
      0,
    ],
    "flowtype/generic-spacing": [
      0,
    ],
    "flowtype/object-type-curly-spacing": [
      0,
    ],
    "flowtype/object-type-delimiter": [
      0,
    ],
    "flowtype/quotes": [
      0,
    ],
    "flowtype/semi": [
      0,
    ],
    "flowtype/space-after-type-colon": [
      0,
    ],
    "flowtype/space-before-generic-bracket": [
      0,
    ],
    "flowtype/space-before-type-colon": [
      0,
    ],
    "flowtype/union-intersection-spacing": [
      0,
    ],
    "for-direction": [
      2,
    ],
    "func-call-spacing": [
      0,
    ],
    "func-names": [
      2,
    ],
    "function-call-argument-newline": [
      0,
    ],
    "function-paren-newline": [
      0,
    ],
    "generator-star": [
      0,
    ],
    "generator-star-spacing": [
      0,
    ],
    "getter-return": [
      2,
      {
        "allowImplicit": true,
      },
    ],
    "gettext/no-variable-string": [
      2,
    ],
    "guard-for-in": [
      2,
    ],
    "implicit-arrow-linebreak": [
      0,
    ],
    "import/default": [
      0,
    ],
    "import/dynamic-import-chunkname": [
      2,
    ],
    "import/export": [
      2,
    ],
    "import/extensions": [
      2,
      {
        "js": "never",
        "json": "always",
        "jsx": "never",
        "ts": "never",
        "tsx": "never",
      },
    ],
    "import/named": [
      2,
    ],
    "import/namespace": [
      2,
    ],
    "import/newline-after-import": [
      1,
    ],
    "import/no-absolute-path": [
      2,
    ],
    "import/no-amd": [
      2,
    ],
    "import/no-duplicates": [
      1,
      {
        "prefer-inline": true,
      },
    ],
    "import/no-dynamic-require": [
      2,
    ],
    "import/no-extraneous-dependencies": [
      2,
    ],
    "import/no-mutable-exports": [
      2,
    ],
    "import/no-named-as-default": [
      1,
    ],
    "import/no-named-as-default-member": [
      1,
    ],
    "import/no-named-default": [
      2,
    ],
    "import/no-unresolved": [
      2,
      {
        "caseSensitive": true,
        "caseSensitiveStrict": false,
        "commonjs": true,
      },
    ],
    "import/no-webpack-loader-syntax": [
      2,
    ],
    "import/order": [
      1,
      {
        "alphabetize": {
          "caseInsensitive": true,
          "order": "asc",
          "orderImportKind": "ignore",
        },
        "distinctGroup": true,
        "groups": [
          [
            "builtin",
            "external",
          ],
          "internal",
          "parent",
          "sibling",
          "index",
        ],
        "named": false,
        "newlines-between": "always",
        "warnOnUnassignedImports": false,
      },
    ],
    "indent": [
      0,
    ],
    "indent-legacy": [
      0,
    ],
    "jsdoc/check-alignment": [
      2,
    ],
    "jsdoc/check-param-names": [
      2,
    ],
    "jsdoc/check-syntax": [
      2,
    ],
    "jsdoc/check-tag-names": [
      2,
    ],
    "jsdoc/implements-on-classes": [
      2,
    ],
    "jsdoc/require-param-name": [
      2,
    ],
    "jsdoc/require-param-type": [
      2,
    ],
    "jsdoc/require-returns-check": [
      2,
    ],
    "jsdoc/require-returns-type": [
      2,
    ],
    "jsx-a11y/alt-text": [
      2,
    ],
    "jsx-a11y/anchor-ambiguous-text": [
      0,
    ],
    "jsx-a11y/anchor-has-content": [
      2,
    ],
    "jsx-a11y/anchor-is-valid": [
      2,
    ],
    "jsx-a11y/aria-activedescendant-has-tabindex": [
      2,
    ],
    "jsx-a11y/aria-props": [
      2,
    ],
    "jsx-a11y/aria-proptypes": [
      2,
    ],
    "jsx-a11y/aria-role": [
      2,
    ],
    "jsx-a11y/aria-unsupported-elements": [
      2,
    ],
    "jsx-a11y/autocomplete-valid": [
      2,
    ],
    "jsx-a11y/click-events-have-key-events": [
      2,
    ],
    "jsx-a11y/control-has-associated-label": [
      0,
      {
        "ignoreElements": [
          "audio",
          "canvas",
          "embed",
          "input",
          "textarea",
          "tr",
          "video",
        ],
        "ignoreRoles": [
          "grid",
          "listbox",
          "menu",
          "menubar",
          "radiogroup",
          "row",
          "tablist",
          "toolbar",
          "tree",
          "treegrid",
        ],
        "includeRoles": [
          "alert",
          "dialog",
        ],
      },
    ],
    "jsx-a11y/heading-has-content": [
      2,
    ],
    "jsx-a11y/html-has-lang": [
      2,
    ],
    "jsx-a11y/iframe-has-title": [
      2,
    ],
    "jsx-a11y/img-redundant-alt": [
      2,
    ],
    "jsx-a11y/interactive-supports-focus": [
      2,
      {
        "tabbable": [
          "button",
          "checkbox",
          "link",
          "searchbox",
          "spinbutton",
          "switch",
          "textbox",
        ],
      },
    ],
    "jsx-a11y/label-has-associated-control": [
      2,
    ],
    "jsx-a11y/label-has-for": [
      0,
    ],
    "jsx-a11y/media-has-caption": [
      2,
    ],
    "jsx-a11y/mouse-events-have-key-events": [
      2,
    ],
    "jsx-a11y/no-access-key": [
      2,
    ],
    "jsx-a11y/no-autofocus": [
      2,
    ],
    "jsx-a11y/no-distracting-elements": [
      2,
    ],
    "jsx-a11y/no-interactive-element-to-noninteractive-role": [
      2,
      {
        "canvas": [
          "img",
        ],
        "tr": [
          "none",
          "presentation",
        ],
      },
    ],
    "jsx-a11y/no-noninteractive-element-interactions": [
      2,
      {
        "alert": [
          "onKeyUp",
          "onKeyDown",
          "onKeyPress",
        ],
        "body": [
          "onError",
          "onLoad",
        ],
        "dialog": [
          "onKeyUp",
          "onKeyDown",
          "onKeyPress",
        ],
        "handlers": [
          "onClick",
          "onError",
          "onLoad",
          "onMouseDown",
          "onMouseUp",
          "onKeyPress",
          "onKeyDown",
          "onKeyUp",
        ],
        "iframe": [
          "onError",
          "onLoad",
        ],
        "img": [
          "onError",
          "onLoad",
        ],
      },
    ],
    "jsx-a11y/no-noninteractive-element-to-interactive-role": [
      2,
      {
        "fieldset": [
          "radiogroup",
          "presentation",
        ],
        "li": [
          "menuitem",
          "menuitemradio",
          "menuitemcheckbox",
          "option",
          "row",
          "tab",
          "treeitem",
        ],
        "ol": [
          "listbox",
          "menu",
          "menubar",
          "radiogroup",
          "tablist",
          "tree",
          "treegrid",
        ],
        "table": [
          "grid",
        ],
        "td": [
          "gridcell",
        ],
        "ul": [
          "listbox",
          "menu",
          "menubar",
          "radiogroup",
          "tablist",
          "tree",
          "treegrid",
        ],
      },
    ],
    "jsx-a11y/no-noninteractive-tabindex": [
      2,
      {
        "allowExpressionValues": true,
        "roles": [
          "tabpanel",
        ],
        "tags": [],
      },
    ],
    "jsx-a11y/no-onchange": [
      0,
    ],
    "jsx-a11y/no-redundant-roles": [
      2,
    ],
    "jsx-a11y/no-static-element-interactions": [
      2,
      {
        "allowExpressionValues": true,
        "handlers": [
          "onClick",
          "onMouseDown",
          "onMouseUp",
          "onKeyPress",
          "onKeyDown",
          "onKeyUp",
        ],
      },
    ],
    "jsx-a11y/role-has-required-aria-props": [
      2,
    ],
    "jsx-a11y/role-supports-aria-props": [
      2,
    ],
    "jsx-a11y/scope": [
      2,
    ],
    "jsx-a11y/tabindex-no-positive": [
      2,
    ],
    "jsx-quotes": [
      0,
    ],
    "key-spacing": [
      0,
    ],
    "keyword-spacing": [
      0,
    ],
    "linebreak-style": [
      0,
    ],
    "lines-around-comment": [
      0,
    ],
    "max-classes-per-file": [
      2,
      1,
    ],
    "max-len": [
      0,
    ],
    "max-statements-per-line": [
      0,
    ],
    "multiline-ternary": [
      0,
    ],
    "new-cap": [
      2,
      {
        "capIsNew": false,
        "capIsNewExceptions": [
          "Immutable.Map",
          "Immutable.Set",
          "Immutable.List",
        ],
        "newIsCap": true,
        "newIsCapExceptions": [],
        "properties": true,
      },
    ],
    "new-parens": [
      0,
    ],
    "newline-per-chained-call": [
      0,
    ],
    "no-alert": [
      1,
    ],
    "no-array-constructor": [
      2,
    ],
    "no-arrow-condition": [
      0,
    ],
    "no-async-promise-executor": [
      2,
    ],
    "no-await-in-loop": [
      2,
    ],
    "no-bitwise": [
      2,
    ],
    "no-caller": [
      2,
    ],
    "no-case-declarations": [
      2,
    ],
    "no-class-assign": [
      2,
    ],
    "no-comma-dangle": [
      0,
    ],
    "no-compare-neg-zero": [
      2,
    ],
    "no-cond-assign": [
      2,
      "always",
    ],
    "no-confusing-arrow": [
      0,
    ],
    "no-console": [
      1,
    ],
    "no-const-assign": [
      2,
    ],
    "no-constant-condition": [
      1,
    ],
    "no-continue": [
      2,
    ],
    "no-control-regex": [
      2,
    ],
    "no-debugger": [
      2,
    ],
    "no-delete-var": [
      2,
    ],
    "no-dupe-args": [
      2,
    ],
    "no-dupe-class-members": [
      2,
    ],
    "no-dupe-else-if": [
      2,
    ],
    "no-dupe-keys": [
      2,
    ],
    "no-duplicate-case": [
      2,
    ],
    "no-duplicate-imports": [
      2,
    ],
    "no-else-return": [
      2,
      {
        "allowElseIf": true,
      },
    ],
    "no-empty": [
      2,
    ],
    "no-empty-character-class": [
      2,
    ],
    "no-empty-function": [
      2,
      {
        "allow": [
          "arrowFunctions",
          "functions",
          "methods",
        ],
      },
    ],
    "no-empty-pattern": [
      2,
    ],
    "no-eval": [
      2,
    ],
    "no-ex-assign": [
      2,
    ],
    "no-extend-native": [
      2,
    ],
    "no-extra-bind": [
      2,
    ],
    "no-extra-boolean-cast": [
      2,
    ],
    "no-extra-label": [
      2,
    ],
    "no-extra-parens": [
      0,
    ],
    "no-extra-semi": [
      0,
    ],
    "no-fallthrough": [
      2,
    ],
    "no-floating-decimal": [
      0,
    ],
    "no-func-assign": [
      2,
    ],
    "no-global-assign": [
      2,
    ],
    "no-implied-eval": [
      2,
    ],
    "no-import-assign": [
      2,
    ],
    "no-inner-declarations": [
      2,
    ],
    "no-invalid-regexp": [
      2,
    ],
    "no-irregular-whitespace": [
      2,
    ],
    "no-iterator": [
      2,
    ],
    "no-label-var": [
      2,
    ],
    "no-labels": [
      2,
      {
        "allowLoop": false,
        "allowSwitch": false,
      },
    ],
    "no-lone-blocks": [
      2,
    ],
    "no-lonely-if": [
      2,
    ],
    "no-loop-func": [
      2,
    ],
    "no-loss-of-precision": [
      2,
    ],
    "no-misleading-character-class": [
      2,
    ],
    "no-mixed-operators": [
      0,
    ],
    "no-mixed-spaces-and-tabs": [
      0,
    ],
    "no-multi-assign": [
      2,
    ],
    "no-multi-spaces": [
      0,
    ],
    "no-multi-str": [
      2,
    ],
    "no-multiple-empty-lines": [
      0,
    ],
    "no-nested-ternary": [
      2,
    ],
    "no-new": [
      2,
    ],
    "no-new-func": [
      2,
    ],
    "no-new-symbol": [
      2,
    ],
    "no-new-wrappers": [
      2,
    ],
    "no-nonoctal-decimal-escape": [
      2,
    ],
    "no-obj-calls": [
      2,
    ],
    "no-object-constructor": [
      2,
    ],
    "no-octal": [
      2,
    ],
    "no-octal-escape": [
      2,
    ],
    "no-param-reassign": [
      2,
      {
        "props": false,
      },
    ],
    "no-plusplus": [
      2,
    ],
    "no-proto": [
      2,
    ],
    "no-prototype-builtins": [
      0,
    ],
    "no-redeclare": [
      2,
    ],
    "no-regex-spaces": [
      2,
    ],
    "no-reserved-keys": [
      0,
    ],
    "no-restricted-globals": [
      2,
      "isFinite",
      "isNaN",
      "addEventListener",
      "blur",
      "close",
      "closed",
      "confirm",
      "defaultStatus",
      "event",
      "external",
      "defaultstatus",
      "find",
      "focus",
      "frameElement",
      "frames",
      "history",
      "innerHeight",
      "innerWidth",
      "length",
      "location",
      "locationbar",
      "menubar",
      "moveBy",
      "moveTo",
      "name",
      "onblur",
      "onerror",
      "onfocus",
      "onload",
      "onresize",
      "onunload",
      "open",
      "opener",
      "opera",
      "outerHeight",
      "outerWidth",
      "pageXOffset",
      "pageYOffset",
      "parent",
      "print",
      "removeEventListener",
      "resizeBy",
      "resizeTo",
      "screen",
      "screenLeft",
      "screenTop",
      "screenX",
      "screenY",
      "scroll",
      "scrollbars",
      "scrollBy",
      "scrollTo",
      "scrollX",
      "scrollY",
      "self",
      "status",
      "statusbar",
      "stop",
      "toolbar",
      "top",
    ],
    "no-restricted-imports": [
      2,
    ],
    "no-restricted-properties": [
      2,
      {
        "message": "arguments.callee is deprecated",
        "object": "arguments",
        "property": "callee",
      },
      {
        "message": "Please use Number.isFinite instead",
        "object": "global",
        "property": "isFinite",
      },
      {
        "message": "Please use Number.isFinite instead",
        "object": "self",
        "property": "isFinite",
      },
      {
        "message": "Please use Number.isFinite instead",
        "object": "window",
        "property": "isFinite",
      },
      {
        "message": "Please use Number.isNaN instead",
        "object": "global",
        "property": "isNaN",
      },
      {
        "message": "Please use Number.isNaN instead",
        "object": "self",
        "property": "isNaN",
      },
      {
        "message": "Please use Number.isNaN instead",
        "object": "window",
        "property": "isNaN",
      },
      {
        "message": "Please use Object.defineProperty instead.",
        "property": "__defineGetter__",
      },
      {
        "message": "Please use Object.defineProperty instead.",
        "property": "__defineSetter__",
      },
      {
        "message": "Use the exponentiation operator (**) instead.",
        "object": "Math",
        "property": "pow",
      },
    ],
    "no-restricted-syntax": [
      2,
      {
        "message": "for..in loops iterate over the entire prototype chain, which is virtually never what you want. Use Object.{keys,values,entries}, and iterate over the resulting array.",
        "selector": "ForInStatement",
      },
      {
        "message": "iterators/generators require regenerator-runtime, which is too heavyweight for this guide to allow them. Separately, loops should be avoided in favor of array iterations.",
        "selector": "ForOfStatement",
      },
      {
        "message": "Labels are a form of GOTO; using them makes code confusing and hard to maintain and understand.",
        "selector": "LabeledStatement",
      },
      {
        "message": "\`with\` is disallowed in strict mode because it makes code impossible to predict and optimize.",
        "selector": "WithStatement",
      },
      {
        "message": "Property setters are not allowed",
        "selector": "MethodDefinition[kind="set"]",
      },
      {
        "message": "Property getters are not allowed",
        "selector": "MethodDefinition[kind="get"]",
      },
    ],
    "no-return-assign": [
      2,
      "except-parens",
    ],
    "no-script-url": [
      2,
    ],
    "no-self-assign": [
      2,
    ],
    "no-self-compare": [
      2,
    ],
    "no-sequences": [
      2,
    ],
    "no-setter-return": [
      2,
    ],
    "no-shadow": [
      2,
      {
        "builtinGlobals": false,
        "hoist": "all",
        "ignoreOnInitialization": false,
      },
    ],
    "no-shadow-restricted-names": [
      2,
    ],
    "no-space-before-semi": [
      0,
    ],
    "no-spaced-func": [
      0,
    ],
    "no-sparse-arrays": [
      2,
    ],
    "no-tabs": [
      0,
    ],
    "no-template-curly-in-string": [
      2,
    ],
    "no-this-before-super": [
      2,
    ],
    "no-trailing-spaces": [
      0,
    ],
    "no-undef": [
      2,
    ],
    "no-undef-init": [
      2,
    ],
    "no-underscore-dangle": [
      2,
      {
        "allowAfterSuper": true,
        "allowAfterThis": true,
        "allowAfterThisConstructor": false,
        "allowFunctionParams": true,
        "allowInArrayDestructuring": true,
        "allowInObjectDestructuring": true,
        "enforceInClassFields": false,
        "enforceInMethodNames": false,
      },
    ],
    "no-unexpected-multiline": [
      0,
    ],
    "no-unneeded-ternary": [
      2,
      {
        "defaultAssignment": false,
      },
    ],
    "no-unreachable": [
      2,
    ],
    "no-unsafe-finally": [
      2,
    ],
    "no-unsafe-negation": [
      2,
    ],
    "no-unsafe-optional-chaining": [
      2,
    ],
    "no-unused-expressions": [
      2,
    ],
    "no-unused-labels": [
      2,
    ],
    "no-unused-vars": [
      2,
      {
        "args": "after-used",
        "ignoreRestSiblings": true,
        "vars": "all",
      },
    ],
    "no-use-before-define": [
      2,
      "nofunc",
    ],
    "no-useless-backreference": [
      2,
    ],
    "no-useless-catch": [
      2,
    ],
    "no-useless-computed-key": [
      2,
    ],
    "no-useless-concat": [
      2,
    ],
    "no-useless-constructor": [
      2,
    ],
    "no-useless-escape": [
      2,
    ],
    "no-useless-rename": [
      2,
      {
        "ignoreDestructuring": false,
        "ignoreExport": false,
        "ignoreImport": false,
      },
    ],
    "no-useless-return": [
      2,
    ],
    "no-var": [
      2,
    ],
    "no-void": [
      2,
      {
        "allowAsStatement": true,
      },
    ],
    "no-whitespace-before-property": [
      0,
    ],
    "no-with": [
      2,
    ],
    "no-wrap-func": [
      0,
    ],
    "nonblock-statement-body-position": [
      0,
    ],
    "object-curly-newline": [
      0,
    ],
    "object-curly-spacing": [
      0,
    ],
    "object-property-newline": [
      0,
    ],
    "object-shorthand": [
      2,
      "always",
      {
        "avoidQuotes": true,
        "ignoreConstructors": false,
      },
    ],
    "one-var": [
      2,
      "never",
    ],
    "one-var-declaration-per-line": [
      0,
    ],
    "operator-assignment": [
      2,
      "always",
    ],
    "operator-linebreak": [
      0,
    ],
    "padded-blocks": [
      0,
    ],
    "prefer-arrow-callback": [
      0,
      {
        "allowNamedFunctions": false,
        "allowUnboundThis": true,
      },
    ],
    "prefer-const": [
      2,
      {
        "destructuring": "any",
        "ignoreReadBeforeAssign": true,
      },
    ],
    "prefer-numeric-literals": [
      2,
    ],
    "prefer-rest-params": [
      2,
    ],
    "prefer-template": [
      2,
    ],
    "prettier/prettier": [
      1,
      {
        "printWidth": 100,
        "singleQuote": true,
        "trailingComma": "all",
      },
      {
        "usePrettierrc": false,
      },
    ],
    "quote-props": [
      0,
    ],
    "quotes": [
      0,
    ],
    "radix": [
      2,
    ],
    "react-hooks/exhaustive-deps": [
      2,
    ],
    "react-hooks/rules-of-hooks": [
      2,
    ],
    "react/destructuring-assignment": [
      2,
    ],
    "react/display-name": [
      0,
    ],
    "react/jsx-child-element-spacing": [
      0,
    ],
    "react/jsx-closing-bracket-location": [
      0,
    ],
    "react/jsx-closing-tag-location": [
      0,
    ],
    "react/jsx-curly-brace-presence": [
      2,
      {
        "children": "never",
        "propElementValues": "always",
        "props": "never",
      },
    ],
    "react/jsx-curly-newline": [
      0,
    ],
    "react/jsx-curly-spacing": [
      0,
    ],
    "react/jsx-equals-spacing": [
      0,
    ],
    "react/jsx-first-prop-new-line": [
      0,
    ],
    "react/jsx-fragments": [
      2,
      "syntax",
    ],
    "react/jsx-indent": [
      0,
    ],
    "react/jsx-indent-props": [
      0,
    ],
    "react/jsx-key": [
      2,
    ],
    "react/jsx-max-props-per-line": [
      0,
    ],
    "react/jsx-newline": [
      0,
    ],
    "react/jsx-no-bind": [
      2,
      {
        "allowArrowFunctions": true,
        "allowBind": false,
        "allowFunctions": false,
        "ignoreDOMComponents": false,
        "ignoreRefs": false,
      },
    ],
    "react/jsx-no-comment-textnodes": [
      2,
    ],
    "react/jsx-no-duplicate-props": [
      2,
    ],
    "react/jsx-no-target-blank": [
      2,
    ],
    "react/jsx-no-undef": [
      2,
    ],
    "react/jsx-no-useless-fragment": [
      2,
    ],
    "react/jsx-one-expression-per-line": [
      0,
    ],
    "react/jsx-pascal-case": [
      2,
    ],
    "react/jsx-props-no-multi-spaces": [
      0,
    ],
    "react/jsx-sort-props": [
      1,
    ],
    "react/jsx-space-before-closing": [
      0,
    ],
    "react/jsx-tag-spacing": [
      0,
    ],
    "react/jsx-uses-react": [
      2,
    ],
    "react/jsx-uses-vars": [
      2,
    ],
    "react/jsx-wrap-multilines": [
      0,
    ],
    "react/no-children-prop": [
      2,
    ],
    "react/no-danger-with-children": [
      2,
    ],
    "react/no-deprecated": [
      2,
    ],
    "react/no-direct-mutation-state": [
      2,
    ],
    "react/no-find-dom-node": [
      2,
    ],
    "react/no-is-mounted": [
      2,
    ],
    "react/no-redundant-should-component-update": [
      2,
    ],
    "react/no-render-return-value": [
      2,
    ],
    "react/no-string-refs": [
      2,
    ],
    "react/no-this-in-sfc": [
      2,
    ],
    "react/no-unescaped-entities": [
      2,
      {
        "forbid": [
          {
            "alternatives": [
              "&gt;",
            ],
            "char": ">",
          },
          {
            "alternatives": [
              "&#125;",
            ],
            "char": "}",
          },
        ],
      },
    ],
    "react/no-unknown-property": [
      2,
    ],
    "react/no-unsafe": [
      2,
    ],
    "react/no-unused-state": [
      2,
    ],
    "react/prefer-es6-class": [
      2,
    ],
    "react/prefer-read-only-props": [
      2,
    ],
    "react/prop-types": [
      0,
    ],
    "react/react-in-jsx-scope": [
      2,
    ],
    "react/require-render-return": [
      2,
    ],
    "react/self-closing-comp": [
      2,
    ],
    "react/style-prop-object": [
      2,
    ],
    "require-yield": [
      2,
    ],
    "rest-spread-spacing": [
      0,
    ],
    "semi": [
      0,
    ],
    "semi-spacing": [
      0,
    ],
    "semi-style": [
      0,
    ],
    "sort-imports": [
      2,
      {
        "allowSeparatedGroups": false,
        "ignoreCase": true,
        "ignoreDeclarationSort": true,
        "ignoreMemberSort": false,
      },
    ],
    "space-after-function-name": [
      0,
    ],
    "space-after-keywords": [
      0,
    ],
    "space-before-blocks": [
      0,
    ],
    "space-before-function-paren": [
      0,
    ],
    "space-before-function-parentheses": [
      0,
    ],
    "space-before-keywords": [
      0,
    ],
    "space-in-brackets": [
      0,
    ],
    "space-in-parens": [
      0,
    ],
    "space-infix-ops": [
      0,
    ],
    "space-return-throw-case": [
      0,
    ],
    "space-unary-ops": [
      0,
    ],
    "space-unary-word-ops": [
      0,
    ],
    "standard/array-bracket-even-spacing": [
      0,
    ],
    "standard/computed-property-even-spacing": [
      0,
    ],
    "standard/object-curly-even-spacing": [
      0,
    ],
    "strict": [
      2,
    ],
    "switch-case/newline-between-switch-case": [
      1,
      "always",
      {
        "fallthrough": "never",
      },
    ],
    "switch-colon-spacing": [
      0,
    ],
    "symbol-description": [
      2,
    ],
    "template-curly-spacing": [
      0,
    ],
    "template-tag-spacing": [
      0,
    ],
    "unicorn/empty-brace-spaces": [
      0,
    ],
    "unicorn/no-nested-ternary": [
      0,
    ],
    "unicorn/number-literal-case": [
      0,
    ],
    "unicorn/template-indent": [
      0,
    ],
    "use-isnan": [
      2,
    ],
    "valid-jsdoc": [
      0,
      {
        "requireParamDescription": false,
        "requireParamType": true,
        "requireReturn": true,
        "requireReturnDescription": false,
        "requireReturnType": true,
      },
    ],
    "valid-typeof": [
      2,
      {
        "requireStringLiterals": true,
      },
    ],
    "vars-on-top": [
      2,
    ],
    "vue/array-bracket-newline": [
      0,
    ],
    "vue/array-bracket-spacing": [
      0,
    ],
    "vue/array-element-newline": [
      0,
    ],
    "vue/arrow-spacing": [
      0,
    ],
    "vue/block-spacing": [
      0,
    ],
    "vue/block-tag-newline": [
      0,
    ],
    "vue/brace-style": [
      0,
    ],
    "vue/comma-dangle": [
      0,
    ],
    "vue/comma-spacing": [
      0,
    ],
    "vue/comma-style": [
      0,
    ],
    "vue/dot-location": [
      0,
    ],
    "vue/func-call-spacing": [
      0,
    ],
    "vue/html-closing-bracket-newline": [
      0,
    ],
    "vue/html-closing-bracket-spacing": [
      0,
    ],
    "vue/html-end-tags": [
      0,
    ],
    "vue/html-indent": [
      0,
    ],
    "vue/html-quotes": [
      0,
    ],
    "vue/html-self-closing": [
      0,
    ],
    "vue/key-spacing": [
      0,
    ],
    "vue/keyword-spacing": [
      0,
    ],
    "vue/max-attributes-per-line": [
      0,
    ],
    "vue/max-len": [
      0,
    ],
    "vue/multiline-html-element-content-newline": [
      0,
    ],
    "vue/multiline-ternary": [
      0,
    ],
    "vue/mustache-interpolation-spacing": [
      0,
    ],
    "vue/no-extra-parens": [
      0,
    ],
    "vue/no-multi-spaces": [
      0,
    ],
    "vue/no-spaces-around-equal-signs-in-attribute": [
      0,
    ],
    "vue/object-curly-newline": [
      0,
    ],
    "vue/object-curly-spacing": [
      0,
    ],
    "vue/object-property-newline": [
      0,
    ],
    "vue/operator-linebreak": [
      0,
    ],
    "vue/quote-props": [
      0,
    ],
    "vue/script-indent": [
      0,
    ],
    "vue/singleline-html-element-content-newline": [
      0,
    ],
    "vue/space-in-parens": [
      0,
    ],
    "vue/space-infix-ops": [
      0,
    ],
    "vue/space-unary-ops": [
      0,
    ],
    "vue/template-curly-spacing": [
      0,
    ],
    "wrap-iife": [
      0,
    ],
    "wrap-regex": [
      0,
    ],
    "yield-star-spacing": [
      0,
    ],
    "yoda": [
      2,
    ],
  },
  "settings": {
    "react": {
      "version": "detect",
    },
  },
}
`;
